### PR TITLE
[QC-430] Avoid clashes when running CCDB tests multiple times in parallel

### PR DESCRIPTION
**Pull request by Barthelemy**

### CODE REVIEW

**Comment by Barthelemy** [comment]
It uses the pid in the Task Name and the CheckName.

**Review by knopers8 — COMMENTED** [review]
Is it fine to have thousands of leftover test objects in test-ccdb? Maybe we should clean them up after the test?

File: Framework/test/testCcdbDatabase.cxx [hunk 1]
```diff
@@ -52,21 +52,45 @@ struct test_fixture {
...
-  f.backend->truncate("my/task", "*");
-}
+struct MyGlobalFixture {
+  void teardown()
+  {
+    std::unique_ptr<CcdbDatabase> backend = std::make_unique<CcdbDatabase>();
+    backend->connect(CCDB_ENDPOINT, "", "", "");
+    // cannot use the test_fixture because we are tearing down
+    backend->truncate("qc/TST/MO/Test/pid", "*");
+    backend->truncate("qc/TST/QO/Test/pid", "*");
```
[the remark this answers is not in the document]
Comment (Barthelemy) [reply Framework/test/testCcdbDatabase.cxx:90]: you are correct, my bad I forgot to add the actual pid. Fixing it.

File: Framework/test/testCcdbDatabase.cxx [hunk 2]
```diff
@@ -52,21 +52,45 @@ struct test_fixture {
...
+  {
+    string fullMoPath = getMoPath(objectName);
+    return fullMoPath.substr(0, fullMoPath.find_last_of('/'));
+  }
+
   std::unique_ptr<CcdbDatabase> backend;
   map<string, string> metadata;
+  const std::string pid = std::to_string(getpid());
+  std::string detector = "TST";
+  std::string taskName = "Test/pid" + pid;
```
[the remark this answers is not in the document]
Comment (Barthelemy) [reply Framework/test/testCcdbDatabase.cxx:80]: I thought so but you have me doubting now. I will put it in the constructor to be fine.

**Comment by Barthelemy** [comment]
> Is it fine to have thousands of leftover test objects in test-ccdb? Maybe we should clean them up after the test?

This is why there is the (faulty, being fixed) truncate, isn't it ?

File: Framework/test/testCcdbDatabase.cxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (knopers8) [reply Framework/test/testCcdbDatabase.cxx:90]: Ah, I see, this invokes it:
```
BOOST_TEST_GLOBAL_FIXTURE(MyGlobalFixture);
```

**Comment by Barthelemy** [comment]
Thank you, I have changed what we discussed above.

**Review by knopers8 — APPROVED** [review]

**Comment by Barthelemy** [comment]
macos builder broken , merging

**Pull request merged**